source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -313,6 +318,26 @@ json TransformerInstance::substring_callback(const inja::Arguments &args) const
   return input.substr(start, substring_len);
 }
 
+json TransformerInstance::replace_with_random_callback(const inja::Arguments &args) {
+    // first argument: string to modify
+  const std::string &source = args.at(0)->get_ref<const std::string &>();
+    // second argument: pattern to be replaced
+  const std::string &to_replace = args.at(1)->get_ref<const std::string &>();
+
+  return 
+    absl::StrReplaceAll(source, {{to_replace, absl::StrCat(random_for_pattern(to_replace))}});
+}
+
+int TransformerInstance::random_for_pattern(const std::string& pattern) {
+  auto found = pattern_replacements_.find(pattern);
+  if (found == pattern_replacements_.end()) {
+    auto random = rng_.random();
```

Comment:
any guarantees here about length and the chance of collisions? i think id slightly prefer the UID version of this func

Reply:
This uses envoy's `Envoy::Random::RandomGeneratorImpl`, which uses BoringSSL's `RAND_bytes` (billed as cryptographically secure pseudo rng, see here: https://www.openssl.org/docs/manmaster/man3/RAND_bytes.html) at its core.

As currently implemented it pre-fetches 256*64 bits of randomness, which it then returns as needed. Both `RandomGeneratorImpl::random()` and `RandomGeneratorImpl::uuid()` consume 64 bits of "randomness" at a time, a probability of collision is then 1/2<sup>64</sup>. 

[Link to `RandomGeneratorImpl`](https://github.com/envoyproxy/envoy/blob/main/source/common/common/random_generator.cc#L12).